storage/2023-11-03/file/shares/create.go

```diff
@@ -62,6 +65,30 @@ func (c Client) Create(ctx context.Context, shareName string, input CreateInput)
 		return
 	}
 
+	// Retry the share creation if a conflicting share is still in the process of being deleted
+	retryFunc := func(resp *http.Response, _ *odata.OData) (bool, error) {
+		if resp != nil {
+			if resp.StatusCode == http.StatusConflict {
```

Comment:
We should be able to make this:

```suggestion
			if response.WasStatusCode(resp, http.StatusConflict) {
```